src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java

```diff
@@ -403,16 +427,24 @@ public boolean isMergeable() {
     }
 
     public String getTarget() {
-        return target;
+        try {
+            return getPullRequest(false).getBase().getRef();
+        } catch (IOException e) {
+            return "UNKNOWN";
+        }
     }
 
     public String getSource() {
-        return source;
+        try {
+            return getPullRequest(false).getHead().getRef();
+        } catch (IOException e) {
+            return "UNKNOWN";
+        }
     }
 
     public String getAuthorEmail() {
```

Comment:
I actually see this call blocking a bit now and then.  It might be worth caching the result